src/app/utils/route-utils.ts

```diff
@@ -148,6 +148,13 @@ export const addressParamLoader = async ({ params }: LoaderFunctionArgs) => {
   return address
 }
 
+export const contractAddressParamLoader = async ({ params }: LoaderFunctionArgs) => {
```

Comment:
you mean move type definition into route utils? useLoaderData returns `unknown` as far as I remember that's why we cast types when we use it.

Reply:
Yeah, it can be moved to RouteUtils, or separate type file. The thing is, if something in loader function changes, since useLoaderData is casted, you won't know there has been any change in the loader function. If that makes sense?

Reply:
Either way, we can leave as is. I will refactor it to use `useTransformToOasisAddress` in https://github.com/oasisprotocol/explorer/pull/1039 so that it only returns one type of the address.

Reply:
works for me. I removed a need to pass eth addr from loader.